lib/excon/socket.rb

```diff
@@ -121,9 +121,15 @@ def connect
         family = @data[:proxy][:family]
       end
 
-      dns_resolver = Resolv::DNS.new
-      dns_resolver.timeouts = @data[:dns_timeouts]
-      resolver = Resolv.new([Resolv::Hosts.new, dns_resolver])
+      resolver = @data[:resolv_resolver] || Resolv.new
+
+      # Deprecated
+      if @data[:dns_timeouts]
```

Comment:
Could you add a printed warning about the deprecation as well, which suggests using resolv_resolvers instead? ie something like https://github.com/excon/excon/blob/85556aeb4af10e94f876a8cbdb764f0377fa0d3a/lib/excon.rb#L76